### PR TITLE
Write test cases for browser component #112

### LINKED ISSUE #112: Write test cases for browser component
Write test cases for the browser component. GuiTest to test if the tabs load the correct web page and if the browser works for different number of tabs. Also write unit test to check if the page replacement algorithm works as intended.

### PR DESCRIPTION
**Pull request by yl-coder**
Fixes #112

### CODE REVIEW

**Comment by HansNewbie** [comment]
`JavaFxThreadingRule` should be in `test` folder. By our convention, the `x` should be in lower case. If `JavaFX` should be treated as one term, then should be `Javafx` instead of `JavaFx`. Not sure which one we should follow.

**Comment by yl-coder** [comment]
Ok, created an util folder for it.

File: src/test/java/address/unittests/BrowserManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ * Tests for the BrowserManager correct behaviours
+ */
+public class BrowserManagerTest {
+
+    @Rule
+    /**
+     * To run test cases on JavaFX thread.
+     */
+    public JavafxThreadingRule javaFxRule = new JavafxThreadingRule();
```
Comment (HansNewbie) [inline src/test/java/address/unittests/BrowserManagerTest.java]: `javafxRule`

File: src/test/java/address/unittests/AddressBookBrowserTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        List<Person> sampleList = TestUtil.generateSampleData();
+        ObservableList<Person> list = FXCollections.observableArrayList();
+        list.add(sampleList.get(0));
+        AddressBookBrowser browser = new AddressBookBrowser(3, list);
+
+        // Loads index 0 of list only.
+        browser.loadProfilePage(list.get(0));
+        assertTrue(list.get(0).equals(browser.getBrowserTabs().get(0).getPerson()));
+        assertFalse(list.get(0).equals(browser.getBrowserTabs().get(1).getPerson()));
+        assertFalse(list.get(0).equals(browser.getBrowserTabs().get(2).getPerson()));
```
Comment (HansNewbie) [inline src/test/java/address/unittests/AddressBookBrowserTest.java]: Shouldn't we assert the person in the tab as `null` here?

Comment (yl-coder) [inline src/test/java/address/unittests/AddressBookBrowserTest.java]: browser.getBrowserTabs().get(1).getPerson() == null?

File: src/test/java/address/unittests/BrowserManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        browser.loadProfilePage(list.get(0));
+
+        //edit person in model
+        list.get(0).setGithubUserName("1");
+        browserManager.handleLocalModelChangedEvent(
+                new LocalModelChangedEvent(new ArrayList<Person>(), new ArrayList<ContactGroup>()));
+        //Check if person in the browser is edited.
+        assertTrue(browser.getBrowserTabs().stream().filter(browserTab
+                -> browserTab.getPerson() != null
+                   && browserTab.getPerson().getGithubUserName().equals("1")).findAny().isPresent());
```
Comment (HansNewbie) [inline src/test/java/address/unittests/BrowserManagerTest.java]: Should assert before changing that `list.get(0).getGithubUserName` is not `1`. If the sample data is ever changed such that the `githubUserName` is `1` even before changing, this test does not prove anything.

File: src/test/java/address/unittests/BrowserManagerTest.java (hunk 1 above)
Comment (yl-coder) [inline src/test/java/address/unittests/BrowserManagerTest.java]: ok, fixed

File: src/test/java/address/unittests/AddressBookBrowserTest.java (hunk 2 above)
Comment (HansNewbie) [inline src/test/java/address/unittests/AddressBookBrowserTest.java]: There is `assertNull` if you want to explore. What I am saying is if only 1 person should be loaded, shouldn't the other two be null? If not, then let's keep it this way.

**Pull request merged**